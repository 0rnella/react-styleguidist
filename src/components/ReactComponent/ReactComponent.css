.root {
	margin-bottom: 3rem;
}

.heading {
	font-size: 2rem;
	font-weight: normal;
}

.sourcePath {
	color: #aaa;
	font-size: 0.8rem;
	margin-bottom: 1rem;
	font-family: monospace;
}

.description {
	margin-bottom: 1rem;
}
