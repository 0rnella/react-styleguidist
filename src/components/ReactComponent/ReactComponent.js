import { Component, PropTypes } from 'react';
import Props from 'components/Props';
import Playground from 'components/Playground';

import s from './ReactComponent.css';

export default class ReactComponent extends Component {
	static propTypes = {
		component: PropTypes.object.isRequired
	}

	renderDescription() {
		let description = this.props.component.props.description;
		if (!description) {
			return null;
		}

		return (
			<div className={s.description}>{description}</div>
		);
	}

	renderExamples() {
		let { examples } = this.props.component;
		if (!examples) {
			return null;
		}

		return examples.map((example, index) => {
			switch (example.type) {
				case 'code':
					return (
						<Playground code={example.content} key={index}/>
					);
				case 'html':
					return (
						<div dangerouslySetInnerHTML={{__html: example.content}} key={index}></div>
					);
			}
		});
	}

	render() {
		let { component } = this.props;
		return (
			<div className={s.root}>
				<h2 className={s.heading}>{component.name}</h2>
				{this.renderDescription()}
				<Props props={component.props}/>
				{this.renderExamples()}
			</div>
		);
	}
}
