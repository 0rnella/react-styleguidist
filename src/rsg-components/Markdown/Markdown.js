import React, { PropTypes } from 'react';
import { compiler } from 'markdown-to-jsx';
import mapValues from 'lodash/mapValues';
import memoize from 'lodash/memoize';
import Styled from 'rsg-components/Styled';
import { styles as linkStyles } from 'rsg-components/Link';

// We’re explicitly specifying Webpack loaders here so we could skip specifying them in Webpack configuration.
// That way we could avoid clashes between our loaders and user loaders.
require('!!../../../loaders/style-loader!../../../loaders/css-loader!highlight.js/styles/tomorrow.css');

// Code blocks with server-side syntax highlight
function Code({ children, className }) {
	const isHighlighted = className && className.includes('lang-');
	if (isHighlighted) {
		return (
			<code className={className} dangerouslySetInnerHTML={{ __html: children }} />
		);
	}
	return (
		<code className={className}>{children}</code>
	);
}
Code.propTypes = {
	children: PropTypes.node,
	className: PropTypes.string,
};

// Custom CSS classes for each tag: <em> → <em className={s.em}> + custom components
const getBaseOverrides = memoize(classes => {
	const styleOverrides = mapValues(classes, value => ({
		props: {
			className: value,
		},
	}));

	return {
		...styleOverrides,
		code: {
			component: Code,
			props: {
				className: classes.code,
			},
		},
	};
});

// Inline mode: replace <p> (usual root component) with <span>
const getInlineOverrides = memoize(classes => {
	const overrides = getBaseOverrides(classes);

	return {
		...overrides,
		p: {
			component: 'span',
			props: {
				className: classes.base,
			},
		},
	};
});

const styles = ({ font, monospace, link, linkHover, border, codeBackground }) => ({
	base: {
		fontFamily: font,
		fontSize: 'inherit',
	},
	para: {
		fontFamily: font,
		fontSize: 'inherit',
		margin: [[0, 0, 15, 0]],
		lineHeight: 1.5,
	},
	a: linkStyles({ link, linkHover }).link,
	h1: {
		composes: '$para',
		fontSize: 36,
		fontWeight: 'normal',
	},
	h2: {
		composes: '$para',
		fontSize: 32,
		fontWeight: 'normal',
	},
	h3: {
		composes: '$para',
		fontSize: 26,
		fontWeight: 'normal',
	},
	h4: {
		composes: '$para',
		fontSize: 21,
		fontWeight: 'normal',
	},
	h5: {
		composes: '$para',
		fontSize: 16,
		fontWeight: 'normal',
	},
	h6: {
		composes: '$para',
		fontSize: 16,
		fontWeight: 'normal',
		fontStyle: 'italic',
	},
	p: {
		composes: '$para',
	},
	ul: {
		composes: '$para',
	},
	ol: {
		composes: '$para',
		listStyleType: 'decimal',
	},
	li: {
		composes: '$base',
		listStyleType: 'inherit',
	},
	input: {
		display: 'inline-block',
		margin: [[0, '0.35em', '0.25em', '-1.2em']],
		verticalAlign: 'middle',
	},
	blockquote: {
		composes: '$para',
		fontSize: 14,
		margin: [[15, 30]],
		padding: 0,
	},
	hr: {
		composes: '$para',
		borderWidth: [[0, 0, 1, 0]],
		borderColor: border,
		borderStyle: 'solid',
	},
	em: {
		composes: '$base',
		fontStyle: 'italic',
	},
	strong: {
		composes: '$base',
		fontWeight: 'bold',
	},
	code: {
		fontFamily: monospace,
		fontSize: 'inherit',
		color: 'inherit',
		background: 'transparent',
		whiteSpace: 'inherit',
	},
	pre: {
		backgroundColor: codeBackground,
		border: [[1, border, 'solid']],
		padding: [[12, 15]],
		fontSize: 12,
		borderRadius: 3,
		whiteSpace: 'pre',
	},
	table: {
		composes: '$para',
		borderCollapse: 'collapse',
	},
	thead: {
		composes: '$hr',
	},
	td: {
		fontFamily: font,
		padding: [[6, 15, 6, 0]],
		fontSize: 14,
	},
	th: {
		composes: '$td',
		fontWeight: 'bold',
	},
});

function Markdown({
	classes,
	text,
	inline,
}) {
	const overrides = inline ? getInlineOverrides(classes) : getBaseOverrides(classes);
	return compiler(text, { overrides });
}

Markdown.propTypes = {
	classes: PropTypes.object.isRequired,
	text: PropTypes.string.isRequired,
	inline: PropTypes.bool,
};

export default Styled(styles)(Markdown);
