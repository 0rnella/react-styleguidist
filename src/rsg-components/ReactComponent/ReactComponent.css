.root {
	margin-bottom: 50px;
	font-size: 16px;
}

.header {
	composes: font from "../../styles/common.css";
	margin-bottom: 20px;
	font-weight: normal;
}

.heading {
	margin: 0 0 6px 0;
	font-size: 36px;
	font-weight: normal;
}

.sourcePath {
	composes: monospace from "../../styles/common.css";
	composes: light from "../../styles/colors.css";
	font-size: 14px;
}

.description {
	composes: font from "../../styles/common.css";
	margin-bottom: 15px;
}

.markdown {
	composes: reset font from "../../styles/common.css";
}

.markdown :global a,
.markdown :global a:link,
.markdown :global a:visited {
	color: #1978c8;
	text-decoration: none;
}

.markdown :global a:hover,
.markdown :global a:focus,
.markdown :global a:active {
	color: #f28a25;
}

.markdown :global p,
.markdown :global ul,
.markdown :global ol,
.markdown :global blockquote,
.markdown :global table,
.markdown :global hr,
.markdown :global h3,
.markdown :global h4,
.markdown :global h5,
.markdown :global h6 {
	margin-top: 0;
	margin-bottom: 15px;
}

.markdown :global h3 {
	font-size: 26px;
	font-weight: normal;
}
.markdown :global h4 {
	font-size: 23px;
	font-weight: normal;
}
.markdown :global h5 {
	font-size: 16px;
	font-weight: bold;
}
.markdown :global h6 {
	font-size: 16px;
	font-weight: normal;
	font-style: italic;
}

.markdown :global code {
	font-family: Consolas, "Liberation Mono", Menlo, monospace;
}

.markdown :global blockquote {
	font-size: 13px;
	margin-top: 0;
	margin-left: 50px;
	margin-right: 50px;
	padding: 0;
}

.markdown :global hr {
	border: 1px solid #e8e8e8;
	border-width: 0 0 1px 0;
}

.markdown :global table {
	border-collapse: collapse;
}
.markdown :global thead {
	border: 1px solid #e8e8e8;
	border-width: 0 0 1px 0;
}
.markdown :global td {
	padding-right: 15px;
	padding-top: 6px;
	font-size: 13px;
}
.markdown :global th {
	padding-right: 15px;
	padding-bottom: 6px;
	text-align: left;
	font-size: 13px;
}
