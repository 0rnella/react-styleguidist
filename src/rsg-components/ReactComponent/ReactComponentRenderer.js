import React from 'react';
import PropTypes from 'prop-types';
import Heading from 'rsg-components/Heading';
import Markdown from 'rsg-components/Markdown';
import JsDoc from 'rsg-components/JsDoc';
import Styled from 'rsg-components/Styled';

const styles = ({ color, fontSize, fontFamily, space }) => ({
	root: {
		marginBottom: space[6],
		fontSize: fontSize.text,
	},
	header: {
		marginBottom: space[3],
	},
	heading: {
		color: color.base,
		marginBottom: space[1],
		fontFamily: fontFamily.base,
		fontSize: fontSize.h4,
		fontWeight: 'normal',
	},
	pathLine: {
		fontFamily: fontFamily.monospace,
		color: color.light,
		fontSize: fontSize.small,
	},
	description: {
		color: color.base,
		marginBottom: space[3],
		fontSize: fontSize.text,
	},
	subsection: {
		marginBottom: space[4],
	},
});

export function ReactComponentRenderer(allProps) {
	const { classes, name, slug, pathLine, description, props, methods, tags, examples } = allProps;
	return (
		<div className={classes.root} id={name + '-container'}>
			<header className={classes.header}>
				<Heading
					id={slug}
					deprecated={!!tags.deprecated}
					slotName="componentToolbar"
					slotProps={allProps}
				>
					{name}
				</Heading>
				<div className={classes.pathLine}>{pathLine}</div>
			</header>
			<div className={classes.description}>
				{description && <Markdown text={description} />}
				<JsDoc {...tags} />
			</div>
			{props &&
				<div className={classes.subsection}>
					<h3 className={classes.heading}>Props</h3>
					{props}
				</div>}
			{methods &&
				<div className={classes.subsection}>
					<h3 className={classes.heading}>Methods</h3>
					{methods}
				</div>}
			{examples}
		</div>
	);
}

ReactComponentRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	tags: PropTypes.object,
	name: PropTypes.string.isRequired,
	slug: PropTypes.string.isRequired,
	pathLine: PropTypes.string.isRequired,
	description: PropTypes.string,
	props: PropTypes.node,
	methods: PropTypes.node,
	examples: PropTypes.node,
	isolated: PropTypes.bool,
	metadata: PropTypes.object.isRequired,
};

ReactComponentRenderer.defaultProps = {
	tags: {},
};

export default Styled(styles)(ReactComponentRenderer);
