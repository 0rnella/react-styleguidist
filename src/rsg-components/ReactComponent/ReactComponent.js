import React, { Component, PropTypes } from 'react';
import Markdown from 'rsg-components/Markdown';
import Props from 'rsg-components/Props';
import Examples from 'rsg-components/Examples';
import ReactComponentRenderer from 'rsg-components/ReactComponent/ReactComponentRenderer';

export default class ReactComponent extends Component {
	static propTypes = {
		highlightTheme: PropTypes.string.isRequired,
		component: PropTypes.object.isRequired,
		sidebar: PropTypes.bool,
	};

	renderDescription(description) {
		if (!description) {
			return null;
		}

		return (
			<Markdown text={description} />
		);
	}

	renderProps(props) {
		if (!props.props) {
			return null;
		}

		return (
			<Props props={props} />
		);
	}

	renderExamples(highlightTheme, examples) {
		if (!examples) {
			return null;
		}

		return (
			<Examples highlightTheme={highlightTheme} examples={examples} />
		);
	}

	render() {
		const { highlightTheme, component, sidebar } = this.props;
		return (
			<ReactComponentRenderer
				name={component.name}
				pathLine={component.pathLine}
				description={this.renderDescription(component.props.description)}
				props={this.renderProps(component.props)}
				examples={this.renderExamples(highlightTheme, component.examples)}
				sidebar={sidebar}
			/>
		);
	}
}
