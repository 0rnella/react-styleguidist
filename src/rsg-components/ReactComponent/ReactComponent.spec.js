import test from 'ava';
import React from 'react';
import Markdown from '../Markdown';
import ReactComponent from './ReactComponent';
import Renderer from './Renderer';

test('should render component properties', () => {
	const component = {
		name: 'Foo',
		pathLine: 'foo/bar.js',
		props: {
			description: 'Bar',
		},
	};

	const RCComponentHOC = ReactComponent(Renderer);
	const RCComponent = new RCComponentHOC({ component });
	const actual = shallow(RCComponent.render());

	expect(actual.node, 'to contain',
		<div>
			<header>
				<h2>Foo</h2>
				<div>foo/bar.js</div>
			</header>
			<div>
				<Markdown text="Bar" />
			</div>
		</div>
	);
});
