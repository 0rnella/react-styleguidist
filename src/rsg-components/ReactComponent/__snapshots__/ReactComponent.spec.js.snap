exports[`test render should render component 1`] = `
<div
  id="Foo-container">
  <header>
    <h2
      id="Foo">
      Foo
    </h2>
    <div>
      foo/bar.js
    </div>
    <div>
      <_class
        href="/">
        ‚Üê Back
      </_class>
    </div>
  </header>
  <div>
    Bar
  </div>
  <div>
    <h3>
      Props
    </h3>
  </div>
  <Playground
    code="<button>OK</button>"
    evalInContext={[Function]}
    index={0}
    name="Component" />
  <_class
    text="Hello *world*!" />
</div>
`;

exports[`test should render component renderer 1`] = `
<_class
  description={
    <_class
      text="Bar" />
  }
  examples={
    <Examples
      examples={
        Array [
          Object {
            "content": "<button>OK</button>",
            "evalInContext": [Function],
            "type": "code",
          },
          Object {
            "content": "Hello *world*!",
            "type": "markdown",
          },
        ]
      }
      name="Foo" />
  }
  name="Foo"
  pathLine="foo/bar.js" />
`;
